src/Specification/Glob.php

```diff
@@ -258,4 +321,38 @@ private static function toRegEx(string $glob) : string
         }
         return $delimiter . '^' . $regex . '$' . $delimiter;
     }
+
+    /** @inheritDoc */
+    public function canBeSatisfiedBySomethingBelow(array $value): bool
+    {
+        $valueSegments = explode('/','/'.$value['path']);
```

Comment:
The amount of magic in this method is quite high. When looking at the code at first sight I was wondering what you were doing here. 
After a while, I understood that you are cutting the parts of a directory path to make them match with the same amount of directory separators? 
But I don't understand why all this is needed.
given:
`src/some/sub/dir/afile.php`
`src/some/other/sub/dir/afile.php`
with a glob `src/some/sub/**/*` would never match to `src/some/other/sub/dir/afile.php` 
so the question here is, could a glob match anything in the current directory? if the directory matches the glob,
it could. Otherwise, it can't?

Or do I completely misunderstand what is going on here?

Reply:
Consider:

`/src/some/*/**/*.php`

If the current directory is `/src/some`, then it is false that "directory matches the glob", and it is also false that the "glob can match anything in the current directory" (as in directly in it), but it can still match something deeper.